src/Coveo.Connectors.EasyFilePusher/ProgramArguments.cs

```diff
@@ -0,0 +1,106 @@
+    /// Definition of the program input arguments.
+    /// </summary>
+    public class ProgramArguments
+    {
+        /// <summary>
+        /// 
+        /// </summary>
+        [Option(Required = true, HelpText = "Cloud environment: " + nameof(CloudEnvironment.Hipaa) + ", " + nameof(CloudEnvironment.Prod) + ", " + nameof(CloudEnvironment.QA) + " or " + nameof(CloudEnvironment.Dev) + ".")]
+        public CloudEnvironment Environment { get; set; }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        [Option(Required = true, HelpText = "Cloud region: " + nameof(CloudRegion.UsEast1) + " or " + nameof(CloudRegion.EuWest1) + ".")]
+        public CloudRegion Region { get; set; }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        [Option(Required = true, HelpText = "ID of the organization in which to push documents.")]
+        public string Organization { get; set; } = "";
+
+        /// <summary>
+        /// 
+        /// </summary>
+        [Option(Required = true, HelpText = "ID of the source in which to push documents.")]
+        public string SourceId { get; set; } = "";
+
+        /// <summary>
+        /// 
+        /// </summary>
+        [Option(Required = true, HelpText = "API key to use.")]
+        public string ApiKey { get; set; } = "";
+
+        /// <summary>
+        /// 
+        /// </summary>
+        [Option(Required = true, HelpText = "Path of the local folder that contains the documents to index.")]
+        public string Folder { get; set; } = "";
+
+        /// <summary>
+        /// 
+        /// </summary>
+        [Option(Default = "*", HelpText = "Wildcard expression for which matching files will be pushed. All files are pushed by default.")]
+        public string Include { get; set; } = "";
+
+        /// <summary>
+        /// 
+        /// </summary>
+        [Option(Default = true, HelpText = "Whether to recursively search in sub-folders for files to push. Sub-folders are searched by default.")]
+        public bool Recursive { get; set; }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        [Option(Default = 10, HelpText = "How many files to push per batch.")]
+        public int BatchSize { get; set; }
+    }
+
+    /// <summary>
+    /// Cloud environments in which documents can be pushed.
+    /// </summary>
+    public enum CloudEnvironment
+    {
+        /// <summary>
+        /// 
+        /// </summary>
+        Hipaa,
+
+        /// <summary>
+        /// 
+        /// </summary>
+        Prod,
+
+        /// <summary>
+        /// 
+        /// </summary>
+        QA,
+
+        /// <summary>
+        /// 
+        /// </summary>
+        Dev
+    }
+
+    /// <summary>
+    /// Cloud regions in which documents can be pushed.
+    /// </summary>
+    public enum CloudRegion
+    {
+        /// <summary>
+        /// 
+        /// </summary>
+        UsEast1,
+
+        /// <summary>
+        /// 
+        /// </summary>
+        EuWest1
```

Comment:
You can add APSouthEast2 right away. It will be announced in two weeks anyway.

Reply:
J'aimerais bien l'ajouter tout de suite dans le Easy File Pusher.  Sauf qu'il faut d'abord l'ajouter dans le PlatformSdk (librarie séparée et avec son propre cycle de release).  Les endpoints présentement définis dans le PlatformSdk sont les suivantes:

        //
        // Summary:
        //     Coveo Push API endpoints.
        public static class Endpoint
        {
            public static class UsEast1
            {
                public const string HIPAA_PUSH_API_URL = "https://apihipaa.cloud.coveo.com/push";
                public const string PROD_PUSH_API_URL = "https://api.cloud.coveo.com/push";
                public const string QA_PUSH_API_URL = "https://apiqa.cloud.coveo.com/push";
                public const string DEV_PUSH_API_URL = "https://apidev.cloud.coveo.com/push";
            }
            public static class EuWest1
            {
                public const string PROD_PUSH_API_URL = "https://api-eu.cloud.coveo.com/push";
                public const string QA_PUSH_API_URL = "https://apiqa-eu.cloud.coveo.com/push";
                public const string DEV_PUSH_API_URL = "https://apidev-eu.cloud.coveo.com/push";
            }
        }

Reply:
@rquirion Tu pourras ouvrir une story pour les ajouter dans le platform SDK pis on viendra ajuster le tool après! Anyway on va en avoir besoin pour le crawling module  en Australie

Reply:
Je viens de créer CT-7032.